app/src/main/java/com/github/wnder/GuessLocationActivity.java

```diff
@@ -314,21 +317,40 @@ public void onCameraMove() {
     private void updateCompassMode(){
         SymbolLayer layer = (SymbolLayer) mapboxMap.getStyle().getLayer(ARROW_LAYER_ID);
         View hotbarView = findViewById(R.id.hotbarView);
+        //Arbitrary value based on the radius to check if we are close enough
+        double referenceDistance = GlobalUser.getUser().getRadius() * 1000 / 100;
 
         if (!compassMode) {
             layer.setProperties(PropertyFactory.visibility(Property.NONE));
             hotbarView.setVisibility(View.INVISIBLE);
 
         } else {
-            //Arbitrary value based on the radius to check if we are close enough
-            if (!(GlobalUser.getUser().getRadius() * 1000 / 100 > guessPosition.distanceTo(picturePosition))) { //compass update
+            double distanceDiff = guessPosition.distanceTo(picturePosition);
+            if (referenceDistance < distanceDiff) { //compass update
                 layer.setProperties(PropertyFactory.visibility(Property.VISIBLE));
                 hotbarView.setVisibility(View.INVISIBLE);
 
             } else { //hotbar update
-                //TODO update hotbar
                 layer.setProperties(PropertyFactory.visibility(Property.NONE));
                 hotbarView.setVisibility(View.VISIBLE);
+
+                double ratio = distanceDiff / referenceDistance;
+
+                ProgressBar bar = (ProgressBar) hotbarView;
+                int barValue = (int) (bar.getMax() - (ratio * bar.getMax()));
+                bar.setProgress(barValue);
+                if (barValue > 75){
+                    bar.setProgressTintList(ColorStateList.valueOf(Color.RED));
+                }
+                else if (barValue > 50 && barValue <= 75){
+                    bar.setProgressTintList(ColorStateList.valueOf(Color.YELLOW));
+                }
+                else if (barValue > 25 && barValue <= 50){
+                    bar.setProgressTintList(ColorStateList.valueOf(Color.CYAN));
+                }
+                else if (barValue <= 25){
+                    bar.setProgressTintList(ColorStateList.valueOf(Color.BLUE));
+                }
```

Comment:
Maybe replace that by  a function "updateHotbar" that could be placed in the MapboxHelper class for better codeclimate.

Reply:
There is now this function in MapBoxHelper!